apps/projects/signals.py

```diff
@@ -93,5 +93,16 @@ def increase_poll_answers_count(sender, instance, created, **kwargs):
 
         insight, _ = ProjectInsight.objects.get_or_create(project=project)
         insight.poll_answers += 1
-        insight.active_participants.add(instance.creator.id)
+        if instance.creator:
+            insight.active_participants.add(instance.creator.id)
+        else:
+            answers = Answer.objects.filter(
```

Comment:
The problem with the current way of counting participants is that it is tied to the post_save signal of Vote and Answer. As a user can have multiple Votes + Answers in a single Poll, we can't just increase `insight.unregistered_participants` by one every time this signal triggers but have to check if there's a vote/answer with the same content_id already, and if there's none (or only one as we don't know if the answer/vote was already saved to the db when `post_save` is called). Not sure if this solution is any good, signals should not be long running and also not sure if this is safe, post_save doesn't guarantee that the model has been written to the database, so this might be slightly inaccurate in some edge cases ? we could wrap it in `transaction.on_commit` but this creates other problems, so I'm not really sure what the  best way forward is. Some ideas could be:
* ignore unregistered users in insights (not great)
* do like in this PR and live with possible inaccuracy
* have an extra model which has project id and content _id to check if the content_id has already participated in the project
* find another way to track poll participants (instead of signal on  Vote and Answer model), probably tricky because insights are in a+ and poll is in a4, but we could probably have a signal in a4 which we connect to a signal here

Reply:
I think I prefer the intermediary model with project and content relations. We can talk about it in a future story.